src/AutoMapper/TypeMap.cs

```diff
@@ -263,7 +263,7 @@ public void Seal(TypeMapRegistry typeMapRegistry, IConfigurationProvider configu
                     .Union(_inheritedMaps)
                     .OrderBy(map => map.MappingOrder).ToArray();
 
-            MapExpression = TypeMapPlanBuilder.BuildMapperFunc(this, configurationProvider, typeMapRegistry);
+            MapExpression = new TypeMapPlanBuilder(configurationProvider, typeMapRegistry, this).CreateMapperLambda();

```

Comment:
They can make a new TypeMapPlanBuilder object, cache it and call it at same time on different threads.  Although looking at the code some more you might not have a problem in this scenario.

It would be a problem if the code wasn't thread safe, which is what I thought it might be given saving the parameters as fields and using them in function calls.

I don't expect people to access this class, but if they did you would want them to just build the function and not worry about the specifics.